### DISCUSSION TITLE
[V5] preload HasOne only get the first item from each page

### DISCUSSION THREAD

**Discussion by fatasy** [adonisjs/core#2432] [Help]
I have this code, it only returns the "type" of the first item on each page. Am I doing something wrong?

_it also happens with the exec method_

```
async index({ request }: HttpContextContract) {
    const { page, perPage, sortColumn, sortType } = request.all()
    const query = Book.query()
      .preload('type')
      .preload('genres')

    if (sortColumn) {
      query.orderBy(sortColumn, sortType)
    }

    const books = await query.paginate(page, perPage)
    return books
  }
```

Book Model hasOne

```
@hasOne(() => BookType, {
    localKey: 'typeId',
    foreignKey: 'id'
  })
  public type: HasOne<typeof BookType>
```

BookType Model  

```
export default class BookType extends BaseModel {
  @column({ isPrimary: true })
  public id: number
  @column()
  public name: string
}
```

**Comment by thetutlage** [comment]
Can you enable query debugging and share the SQL queries being executed?

**Reply by fatasy** [reply]
thanks for the answer, this is the result.

**using the exec method**

```
knex:client acquired connection from pool: __knexUid1 +0ms
  knex:query select `books`.`id`, `cover`, `slug`, `title`, `description`, `chapters`, `words`, `type_id` from `books` undefined +0ms
  knex:bindings [] undefined +0ms
  knex:client releasing connection to pool: __knexUid1 +8ms
  knex:client acquired connection from pool: __knexUid1 +5ms
  knex:query select * from `book_types` where `id` in (?, ?) undefined +11ms
  knex:bindings [ 2445, 2100 ] undefined +11ms
  knex:client releasing connection to pool: __knexUid1 +4ms
  knex:client acquired connection from pool: __knexUid1 +1ms
  knex:query select `genres`.*, `book_genre`.`book_id` as `pivot_book_id`, `book_genre`.`genre_id` as `pivot_genre_id` from `genres` inner join `book_genre` on `genres`.`id` = `book_genre`.`genre_id` where `book_genre`.`book_id` in (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) undefined +4ms
```
_print_

![image](https://user-images.githubusercontent.com/18352352/113042827-05878980-9172-11eb-8a01-6c6132046a79.png)

**Reply by thetutlage** [reply]
I mean debug it using the Debugging strategy of AdonisJS. Knex output is hardly readable. Here's the docs on same https://preview.adonisjs.com/guides/database/debugging-queries

**Reply by fatasy** [reply]
sorry I did all the steps, but the logs don't show up. I noticed that this only happens when the result has already been obtained.

for example if the type_id "2445" was already loaded in a previous result it will iginorate the current one, find this out by modifying the type_id field manually in the database.
![aUAW51617301136](https://user-images.githubusercontent.com/18352352/113334341-f7fd0b80-92f9-11eb-8c40-c3887939ab23.png)

sorry for my English, it's not my primary language.

here are the versions of the packages i am using:
```
"@adonisjs/assembler": "^3.0.0",
"@adonisjs/core": "~5.0.4-preview-rc",
"@adonisjs/auth": "^5.1.1",
"@adonisjs/lucid": "^10.0.0",
"@adonisjs/mail": "^5.2.3",
"@adonisjs/repl": "^1.0.0",
"@adonisjs/view": "^3.1.0",
```

**Reply by thetutlage** [reply]
Alright. Can you please share a repo that I can use to reproduce the error?

**Reply by fatasy** [reply]
> Alright. Can you please share a repo that I can use to reproduce the error?

here is https://github.com/fatasy/adonisjs-books-preload, I am using a mysql database, the project already has the migrations and seeders.

thank you very much for your support.

**Reply by fatasy** [reply]
I also tested this project, with a clean install, and the same thing happened.

**Reply by thetutlage** [reply]
Lemme check and get back to you

**Reply by fatasy** [reply]
Did you manage to check this problem?

**Reply by thetutlage** [reply]
Ahhhh. Sorry, kind of slipped from my mind. Lemme give it try now

**Reply by thetutlage** [reply]
I see a few issues with migrations too. The `genres` table is created after the `book_genres` table. But referenced as a FK. So I am not able to get the migrations running

**Reply by thetutlage** [reply]
Also, you say. The book `hasOne` bookType. Whereas the FK are set incorrectly in the database. It should be `belongsTo` relationship, or you should remove `type_id` from the `books` table and define `book_id` in the `book_types` table.

**Reply by fatasy** [reply]
Thanks for the help! I will do this way then.

I just found it strange because it started happening when I migrated to version 5, I have this same project in version 4 and it works. But I think I was doing it wrong all this time, and in version 5 this is no longer allowed.
